src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs

```diff
@@ -29,25 +29,45 @@ public ISchema DeriveSchema(GraphSchemaInfo schemaInfo)
             var interfaces = types
                 .Where(t => t.IsInterfaceType && !t.IsInputType && t.PossibleTypes.Any())
                 .GroupBy(t => t.Name)
-                .Select(g => g.First());
-            var possibleTypes = interfaces
+                .Select(g => g.First());                
+            var myPossibleTypes = interfaces
                 .Where(t => !t.IsIgnored)
                 .SelectMany(t => t.PossibleTypes)
-                .GroupBy(t => t.Name)
-                .Select(g => g.First());
+                .Select(x => OmitNonRootTypes(x))
+                .Where(x => x != null)
+                .ToArray();
```

Comment:
By not grouping, isn't there a chance that you get duplicate references here if the same type appear as "possible type" for more than one interface?